### DISCUSSION TITLE
Pillow 10.0.0 throw "ImageDraw.textsize{}" error

### DISCUSSION THREAD

**Discussion by Lord-ZuzurNC** [netbox-community/netbox#13595] [Help Wanted!] [answered]
Hello,

Firstly I am sorry for not providing that much information, I'm here to ask how I can provide them firstly.

So my env is:
- OS: Ubuntu 22.04 LTS
- Fresh new installation of Netbox 3.5.8 (I followed the netbox.dev documentation to install and didn't changed anything (but pwd) from the suggested conf).
- I installed lots of plugin including: 
``` bash
netbox-topology-views
netbox-documents
netbox_inventory
phonebox-plugin
netbox-qrcode
netbox-ipcalculator
netbox-ipv4-tools
netbox-device-view
netbox-nagios
git+https://github.com/arthanson/netbox_healthcheck_plugin      #This is netbox_healthcheck_plugin
```
- All dependencies are installed (including, but not all) Python3.
- I have, as stated from my title, Pillow 10.0.0 pipy install
``` bash
(venv) netbox@netbox-test:/opt/netbox$ python3 -m pip list -v | grep Pillow
Pillow                        10.0.0       /opt/netbox-3.5.8/venv/lib/python3.10/site-packages pip
```

So when I go to the Netbox WebUI > DCIM > Rack > Then click on one of my Rack (for example) I got that error
![image](https://github.com/netbox-community/netbox/assets/90102166/5cac5d83-ed64-41d0-b39a-9f0b70dac8a2)

I tried to do a grep into all files in "/opt/netbox/" to search all occurrences of "ImageDraw" but got lots of results, attached here:
[Pastebin](https://pastebin.com/N7HwvyTT) cause very long, sorry.

I searched on internet and found out that PIllow 10.0.0 has removed that "ImageDraw.textsize" attribute and so workaround is 9.5.0 or to modify the source code to use something else like "textbox".

I know how to work around:
``` bash
(venv) netbox@netbox-test:/opt/netbox$ pip install Pillow==9.5.0
(venv) netbox@netbox-test:/opt/netbox$ sudo systemctl restart netbox
```

It then no more error. But this error means there is somewhere an "ImageDraw.textsize" depreciated usage.

I think it can come from a plugin but how to know which one is the culprit? Or maybe is it from Netbox and I misread something that state we need to have Pillow 9.5.0 max and not 10.0.0?

Thank you in advance for your helps.

Best Regards
LordZ

**Comment by kkthxbye-code** [comment, marked as answer]
netbox-qrcode

**Reply by Lord-ZuzurNC** [reply]
Hello, 

Thank you for your reply. I will then create a ticket on the appropriate GitHub.

May I ask how did you find out it was netbox-qrcode? Is it known issue?

Thanks again.

**Reply by kkthxbye-code** [reply]
Yes you are the sixth person to create an issue/discussion.

**Reply by Lord-ZuzurNC** [reply]
Oh I just found out the netbox-qrcode maintener just release a 0.0.10 version which mention "Fix Pillow version to use".

I will update the plugin. And if any I will create an issue on his GitHub. So I think this can be closed.
